### PR TITLE
Pagination

### LINKED ISSUE #269: Pagination API Changes
### Component in question:
- `Pagination`

### Description of issue:
- `Pagination` is still using `dropdownOptions`, which has been deprecated with #253. Pagination needs to be upgraded with the next release

### PR DESCRIPTION
**Pull request by helenvholmes**
Issue number: #269 

## **This PR does the following:**
- Updates `Pagination` to use sub components as children and props

### Front End Review:
- [ ] View [the example in Storybook](https://deploy-preview-323--stoic-murdock-c7f044.netlify.app/?path=/story/pagination--pagination)

### CODE REVIEW

File: src/components/Pagination/Pagination.stories.tsx [hunk 1]
```diff
@@ -1,20 +1,78 @@
...
+import Select from "../Select/Select";
 
 export default {
     title: "Pagination",
     component: Pagination,
+    decorators: [withDesign],
 };
 
 export const pagination = () => (
     <Pagination
```
Comment (hokei) [inline src/components/Pagination/Pagination.stories.tsx:21]: I'm like "I can't believe we didn't do this from the start" every time I see this

File: src/components/Pagination/Pagination.stories.tsx [hunk 2]
```diff
@@ -1,20 +1,78 @@
...
-        currentValue={"1 of 4"}
-        onSelectChange={action("selectChange")}
-        onSelectBlur={action("selectBlur")}
-    />
+        previousPage={
+            <Button
+                buttonType={ButtonTypes.Secondary}
+                onClick={action("Previous clicked")}
+                iconDecorative={true}
+                iconName={IconNames.arrow}
```
Comment (hokei) [inline src/components/Pagination/Pagination.stories.tsx:27]: Button still needs to be refactored?

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -1,67 +1,34 @@
...
-export default function Pagination(props: PaginationProps) {
+export default function Pagination(
+    props: React.PropsWithChildren<PaginationProps>
+) {
     const {
         blockName,
+        children,
         className,
-        currentValue,
+        id,
```
Comment (hokei) [inline src/components/Pagination/Pagination.tsx]: is id used?

Comment (hokei) [reply src/components/Pagination/Pagination.tsx]: same with blockName, className, modifiers

Comment (hokei) [reply src/components/Pagination/Pagination.tsx]: Should also add tests that the component does apply id, className, etc through if they're passed in.

File: src/components/Pagination/Pagination.stories.tsx [hunk 4]
```diff
@@ -1,20 +1,78 @@
 import * as React from "react";
 import { action } from "@storybook/addon-actions";
+import { withDesign } from "storybook-addon-designs";
```
Comment (hokei) [inline src/components/Pagination/Pagination.stories.tsx:3]: Is the lack of knobs intended (Eg consumers aren't ever supposed to use this component with button/label text that's different?)

**Review by hokei — CHANGES_REQUESTED** [review]
Looks great!  A few minor things.  

I know this is the `<Select>` component but it's relevant here - I think it couldn't hurt to make `<Select>` throw a console.warn when you try to pass it more than 7 options.  So that if it's going of the page, etc.  the consumer knows they need to do some extra work / use the not-yet-built giant-paginator.

File: src/components/Pagination/Pagination.tsx (hunk 3 above)
Comment (helenvholmes) [reply src/components/Pagination/Pagination.tsx]: I've removed id, but blockName, className, and modifiers allow consumers to change aspects of Pagination for their app and shouldn't be removed.

I think that making sure blockName, className, and modifiers are applied to components might belong better with the BEM helper, OR we should add that test to every component in the library.

Comment (hokei) [reply src/components/Pagination/Pagination.tsx]: if a blockName gets passed in here, where is it applied?  Sorry I feel like I'm missing something very obvious

Comment (helenvholmes) [reply src/components/Pagination/Pagination.tsx]: L33, no?

File: src/components/Pagination/Pagination.stories.tsx (hunk 4 above)
Comment (helenvholmes) [reply src/components/Pagination/Pagination.stories.tsx:3]: I'll add these in!

**Comment by helenvholmes** [comment]
@hokei That's a good suggestion—added!

File: src/components/Pagination/Pagination.stories.tsx (hunk 2 above)
Comment (helenvholmes) [reply src/components/Pagination/Pagination.stories.tsx:27]: Yeah, looks like it

File: src/components/Pagination/Pagination.tsx (hunk 3 above)
Comment (hokei) [reply src/components/Pagination/Pagination.tsx]: GAAAAH github hid literally just these two lines T_T.  Yes is perfect.

**Review by hokei — APPROVED** [review]
Optional:  A test that checks that the warn gets thrown.  But looks great.

**Comment by helenvholmes** [comment]
@hokei that is a great suggestion... that for the life of me I can't figure out how to do. 

I've mocked out an "it-should" statement and will file an issue if someone else wants to try and take that on (I suspect there are a number of people who would find such a thing quite easy).

**Pull request merged**